### PR TITLE
Add unit when formatting attribute for display

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Add unit when formatting attribute for display. 
It also improve the developer experience because we don't have to worry of which unit to use.
It will also be possible to display the unit in the automation editor (not included in this PR) : 
`When Brightness from Bed Light is above 10%` instead of `When Brightness from Bed Light is above 10`

### Example : entities card

```yaml
type: entities
entities:
  - type: attribute
    attribute: temperature
    entity: climate.ac
  - type: attribute
    attribute: battery_level
    entity: vacuum.5_fifth_floor
```

**Before**
CleanShot 2023-08-17 at 16 47 28

**After**
CleanShot 2023-08-17 at 16 47 13

### Notice

For entity card, there is ab option to provide a custom unit so I don't add support for automatic unit formatting to this card.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/data/entity_attributes.ts [hunk 1]
```diff
@@ -21,3 +21,32 @@ export const STATE_ATTRIBUTES = [
...
+  percentage: "%",
+  current_position: "%",
+  current_tilt_position: "%",
+  color_temp: "mired",
+  max_mireds: "mired",
+  min_mireds: "mired",
+  color_temp_kelvin: "K",
+  min_color_temp_kelvin: "K",
+  max_color_temp_kelvin: "K",
+  elevation: "°",
```
Comment (bramkragten) [inline src/data/entity_attributes.ts:51]: wouldn't/couldn't this be distance?

File: src/data/entity_attributes.ts [hunk 2]
```diff
@@ -21,3 +21,32 @@ export const STATE_ATTRIBUTES = [
...
+
+export const TEMPERATURE_ATTRIBUTES = new Set([
+  "temperature",
+  "current_temperature",
+  "target_temperature",
+  "target_temp_temp",
+  "target_temp_high",
+]);
+
+export const ATTRIBUTES_UNITS: Record<string, string> = {
```
Comment (bramkragten) [inline src/data/entity_attributes.ts:33]: Should we maybe make this a list per domain, and only do it for standard attributes of that domain? It feels a bit error prone like this?

File: src/data/entity_attributes.ts (hunk 1 above)
Comment (piitaya) [reply src/data/entity_attributes.ts:51]: Another reason to split the list by domain 😊

File: src/common/entity/compute_attribute_display.ts [hunk 3]
```diff
@@ -31,7 +39,35 @@ export const computeAttributeValueDisplay = (
...
+    if (unit === "°") {
+      return `${formattedValue}${unit}`;
+    }
+
+    if (unit) {
+      return `${formattedValue} ${unit}`;
+    }
+
+    if (TEMPERATURE_ATTRIBUTES.has(attribute)) {
+      return `${formattedValue} ${config.unit_system.temperature}`;
```
Comment (bramkragten) [inline src/common/entity/compute_attribute_display.ts:67]: Is this correct? As we have entity settings for the temperature unit too? Should this use the entity UOM if it is als temperature? 😅 Or is the unit of attribute values not changeable yet?

Comment (piitaya) [reply src/common/entity/compute_attribute_display.ts:66]: We can not change it. We used everywhere `config.unit_system.temperature` before this PR.
There is a special case for weather entity but for climate and water_heater it's not editable.
Do you have any other domain that have temperature attribute?

Comment (bramkragten) [reply src/common/entity/compute_attribute_display.ts:66]: > Do you have any other domain that have temperature attribute?
I don't know, I think it is fine, people should use sensors for all other things anyway. I can imagine for climate and weather we will introduce a setting for units too, then I think we should check the entity reg

**Comment by piitaya** [comment]
There is a special case for brightness that convert the 0-255 value to percentage.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**